x-pack/platform/plugins/shared/onechat/server/services/agents/modes/default/state.ts

```diff
@@ -24,14 +25,23 @@ export const StateAnnotation = Annotation.Root({
     reducer: (a, b) => b,
     default: () => 0,
   }),
-  nextMessage: Annotation<AIMessage>(),
-  maxCycleReached: Annotation<boolean>(),
-  handoverNote: Annotation<string>(),
-  // outputs
-  addedMessages: Annotation<BaseMessage[]>({
-    reducer: messagesStateReducer,
+  // counter to keep track of the number of successive errors
+  errorCount: Annotation<number>({
+    reducer: (a, b) => b,
+    default: () => 0,
+  }),
+  // list of actions/steps performed by the main agent
+  mainActions: Annotation<ResearchAgentAction[]>({
+    reducer: (a, b) => [...a, ...b],
+    default: () => [],
+  }),
+  // list of actions/steps performed by the answer agent
+  answerActions: Annotation<AnswerAgentAction[]>({
+    reducer: (a, b) => [...a, ...b],
     default: () => [],
   }),
```

Comment:
keeping researcher and answer "actions" dissociated, simpler for action-based logic, and also because I suspect we may fork those into 2 subgraphs at some point (was close to doing so in this PR already)